tcping.go

```diff
@@ -535,6 +546,14 @@ func (tcpStats *stats) handleConnSuccess(rtt float32, now time.Time) {
 		tcpStats.wasDown = false
 		tcpStats.ongoingUnsuccessfulProbes = 0
 		tcpStats.ongoingSuccessfulProbes = 0
+
+		lastAddr := tcpStats.hostnameChanges[len(tcpStats.hostnameChanges)-1].Addr
+		if lastAddr != tcpStats.ip {
+			tcpStats.hostnameChanges = append(tcpStats.hostnameChanges, hostnameChange{
+				Addr: tcpStats.ip,
+				When: time.Now(),
+			})
+		}
```

Comment:
This has to go inside the `retryResolve` function since it will only be used when the `-r` flag is provided and the actual IP change happens there. 

That way, the code would be more coherent.

Reply:
Understandable, will fix

Reply:
> We only want to add addresses that have been successfully resolved. However, to determine if the IP is available or not, we need to ping it. That's why the function is inside the handleConnSuccess. If it were inside the resolve function, we wouldn't have the necessary information to check whether it's a "good" address or not.

True that. Although we do not need to validate if the new address worked. We would just need to report all the changes to the user. For example if a company is testing their GSLB, what they are interested is the IP address changes and the interval between them rather than a successful probe.

I know it might sound a bit confusing. 😄 


> (I recently learned that you can use diagrams directly inside gh comments so let's try it)
Here's our lifecycle (simplified)

wow! this was really cool 😮 Look at all the controls is also embeds in.

Reply:
> True that. Although we do not need to validate if the new address worked. We would just need to report all the changes to the user. For example if a company is testing their GSLB, what they are interested in is the IP address changes and the interval between them rather than a successful probe.

Oh I see. I'll move the logic to `retryResolve` then.